src/server/game/map/index.ts

```diff
@@ -462,11 +474,26 @@ export class Map {
     );
   }
 
+  newBarbarianCampAt(coords: Coords): number | null {
+    const tile = this.getTile(coords);
+    if (!this.canSettleOn(tile)) return null;
+
+    const cityID = this.cities.length;
+    const camp: BarbarianCamp = new BarbarianCamp(cityID, coords);
+    this.cities.push(camp);
+
+    this.setTileOwner(coords, camp, false);
+
+    this.buildImprovementAt(coords, 'barbarian_camp');
+    return cityID;
+  }
+
   settleCityAt(coords: Coords, name: string, civID: number, settler: Unit): boolean {
     const tile = this.getTile(coords);
     if (!this.canSettleOn(tile)) return false;
 
-    const city: City = new City(coords, name, civID);
+    const cityID = this.cities.length;
+    const city: City = new City(cityID, coords, name, civID);
     this.cities.push(city);
```

Comment:
Maybe these three lines should be moved into a `makeCity` function?

Reply:
There are actually several places where I found myself thinking similar things - so much that I think we should make that its own ticket, to be honest.

Reply:
Most of the code right now isn't written to be controlled from anywhere else than `methods.ts`. This feels arguably wrong, and we should probably add a lot more `game` and/or `world` level functions to handle the things that `methods.ts` currently handles.

Reply:
This would be covered by #170